src/metabase/lib/js.cljs

```diff
@@ -1238,10 +1238,9 @@
         clj->js)))
 
 (defn- legacy-ref->pMBQL [a-legacy-ref]
-  (-> a-legacy-ref
-      (js->clj :keywordize-keys true)
-      (update 0 keyword)
-      lib.convert/->pMBQL))
+  (->> (-> a-legacy-ref (js->clj :keywordize-keys true) (update 0 keyword))
+       (mbql.normalize/normalize-fragment nil)
+       (lib.convert/->pMBQL)))
```

Comment:
```suggestion
  (-> a-legacy-ref
      (js->clj :keywordize-keys true)
      (update 0 keyword)
      (->> (mbql.normalize/normalize-fragment nil))
      lib.convert/->pMBQL))
```

*Embed `->>` into a `->` with this one weird trick!*

It works with `as->` too if the threaded value needs to go somewhere more complicated:
```clojure
(-> ...
    (as-> $foo (some-fn (calls 1 $foo 3) 4))
    ...)
```

Also, no parens around simple function calls in `->` (like `lib.convert/->pMBQL` here).